### PR TITLE
Add just some simple way to filter out tokens

### PR DESCRIPTION
**Pull request by anxolin**
For Arbitrum launch we have one issue, most of the bridged tokens don't have liquidity.

This creates a bad user experience.

This PR allows the script generators to provide a filtering function that will have the last call on which token should be included and which don't.

In this simplistic implementation, I just allow to cherry-pick a bunch of tokens by token symbol. I know..., token symbols are not unique, but this comes from know lists so its not expected to have duplicates, and the configuration feels simpler to read if I can select the tokens to include by token symbol

## Important
This PR should not be merged, and the token selection is not final. Chen is deciding on the final list of tokens to include, tomorrow he will provide them, so we can launch with a filtered list.

## Edit 04/06/2024
I did a bunch of enhancements for the launch. Some of them need to be iterated later.

The goal was to reduce the list to a bunch of meaningful tokens (chen provided the list)

The problem, some of these were not in our list
Another problem was that these tokens provided by chen in a CSV didn't have a lot of the information (token name, decimals, etc)

This PR added:
- Possibility to add a list of tokens not in the reference list (so you can add tokens even if they are not in coingecko, this way we add the ones picked by chen)
- Utility to read the token list from a CSV
- Get ERC20 info from onchain, so we can enhance the tokens
- Utilities to merge token information, this way we can merge CSV, the token list, and the on-chain data properly
- Additional filters: for it was preferred a smaller list of meaningful tokens (instead of a big list of illiquid tokens), therefore an additional filter was provided (the implementation is, that any token not in the CSV is discarded)

### CODE REVIEW

File: src/scripts/generateBridgeList.ts [hunk 1]
```diff
@@ -34,15 +46,8 @@ type Extensions = {
...
  */
-export async function generateBridgedList(
-  chainId: SupportedChainId,
-  tokenListSource: string | TokenList,
-  bridgeContractAddress: string,
-  bridgeContractAbi: ContractInterface,
-  methodName: string,
-  outputFilePath: string,
-  tokensToReplace: Record<string, string | null> = {},
-): Promise<void> {
```
Comment (anxolin) [inline src/scripts/generateBridgeList.ts:45]: it had too many arguments, so I decided to change from positional parameters to named parameters

File: src/scripts/generateBridgeList.ts [hunk 2]
```diff
@@ -66,13 +71,18 @@ export async function generateBridgedList(
     )
     .then((results) => parseBridgedAddressesResults(results, fetchBridgedAddressesCalls, tokensToReplace, chainId))
 
-  const tokens = await multicall.callStatic
+  const allTokens = await multicall.callStatic
     .tryAggregate(false, getTotalSupplyCalls(bridgedAddresses, provider))
     .then((results) => parseTotalSupplyResponses(results, bridgedAddresses, chainId))
 
+  const tokens = tokenFilter ? allTokens.filter(tokenFilter) : allTokens
```
Comment (anxolin) [inline src/scripts/generateBridgeList.ts:78]: gist of the PR, the function receives now an optional parameter with the actual filter

File: src/scripts/const/arbitrum.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+export const ARBITRUM_BRIDGE_ADDRESS = '0x09e9222e96e7b4ae2a407b98d48e330053351eee'
+
+/**
+ * Most of the tokens that are bridged to Arbitrum using the bridge, they don't have liquidity, so the trade experience is not good.
+ * This list is meant to limit the tokens shown to the ones that have liquidity.
+ * 
+ * Future versions of the script should decide using objective on-chain data if a token is liquid or not.
+ */
+export const TOKENS_WITH_LIQUIDITY = [
+  'ARB', 'cbETH', 'GRT', 'LINK', 'USDC', 'USDT', 'WBTC', 'WETH',
```
Comment (shoom3301) [inline src/scripts/const/arbitrum.ts:10]: Does it mean that we are going to have only 8 tradable tokens on Arbitrum?
Looks a bit poor

Comment (shoom3301) [reply src/scripts/const/arbitrum.ts:10]: We have liquidity for COW but it's not in the list

Comment (anxolin) [reply src/scripts/const/arbitrum.ts:10]: no no, this is a placeholder for tomorrow. 

Chen is working on the actual list. There will be many more

**Pull request merged**